src/ray/raylet/node_manager.cc

```diff
@@ -1287,6 +1287,9 @@ void NodeManager::TreatTaskAsFailedIfLost(const Task &task) {
         [this, task_marked_as_failed, task](
             const ray::ObjectID &object_id,
             const std::unordered_set<ray::ClientID> &clients, bool has_been_created) {
+          // TODO(raulchen): LookupLocations will set has_been_created to false if
+          // the object does not exist on any nodes due to eviction
+          has_been_created = true;
```

Comment:
Hmm this is a bit ugly. What if we kept the key in Redis even for objects that have been evicted from all nodes? Then, we could at least match the current behavior.

Another possibility (which I prefer) is to not rely on the `has_been_created` flag at all, and maybe just put an exception if: (a) `LookupLocations` returns empty, and (b) the object is not local.

Reply:
I prefer to not reply on `has_been_created` as well as well. 

I thought of a case that depending on it would have problems. Let's say an actor finishes a task and saves the result to object store, then it does a checkpoint. Then the node dies before object info is saved to GCS. And the actor gets reconstructed on another node and resumes from the checkpoint. In this case, we cannot re-execute the task and the object would appear like never have been created.

Reply:
Removed `has_been_created` from some callbacks and parameter of `UpdateObjectLocations`.